components/script/dom/response.rs

```diff
@@ -415,7 +415,8 @@ impl Response {
     #[allow(unrooted_must_root)]
     pub fn finish(&self, body: Vec<u8>) {
         *self.body.borrow_mut() = NetTraitsResponseBody::Done(body);
-        if let Some((p, body_type)) = self.body_promise.borrow_mut().take() {
+        let body_promise = self.body_promise.borrow_mut().take();
+        if let Some((p, body_type)) = body_promise {
```

Comment:
It's interesting that all those XHR tests are timing-out, since the promise would previously be traced while `consume_body_with_promise` is called, could it be that it now is garbage collected instead, since it's not rooted anywhere?

Reply:
It doesn't look like this is used by XHR, and it might be something else that either is not traced, or that is `None` and somehow called back into from the script code?

Reply:
The tests are timing out because debugmozjs builds are much slower than release builds, and the gczeal settings are forcing a full GC every 20 JS allocations.